File: week1/pset/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_number_sign(int);

int main(void)
{
    // Prompt for Height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    // Print pyramids of blocks

    for (int i = 1; i <= n; i++)
    {
        int spaces = n - i;
        // Print spaces
        while (spaces--)
        {
            printf(" ");
        }
        print_number_sign(i);
        printf("\n");
    }

    return 0;
}
void print_number_sign(int h)
{
    // Print number_sign
    while (h--)
    {
        printf("#");
    }
}

File: week6/pset/sentimental-mario-more/mario.py
# Prompt the user for height
def get_height():
    while True:
        try:
            n = int(input("Height: "))
            if n > 0 and n < 9:
                return n
        except ValueError:
            print("Not a integer")


# Print the double half-pyramid
height = get_height()
for i in range(height):
    blocks = i + 1
    spaces = height - 1 - i
    print(spaces * " " + blocks * "#" + "  " + blocks * "#")
